src/providers/oauth-provider.js

```diff
@@ -56,22 +50,32 @@ function OAuthProvider() {
     });
 
     // Remove `baseUrl` trailing slash.
-    if('/' === config.baseUrl.substr(-1)) {
+    if ('/' === config.baseUrl.substr(-1)) {
       config.baseUrl = config.baseUrl.slice(0, -1);
     }
 
     // Add `grantPath` facing slash.
-    if('/' !== config.grantPath[0]) {
+    if ('/' !== config.grantPath[0]) {
       config.grantPath = `/${config.grantPath}`;
     }
 
     // Add `revokePath` facing slash.
-    if('/' !== config.revokePath[0]) {
+    if ('/' !== config.revokePath[0]) {
       config.revokePath = `/${config.revokePath}`;
     }
 
     return config;
   };
+  
+  /**
+   * Configure.
+   *
+   * @param {object} params - An `object` of params to extend.
+   */
+  this.configure = function(params) {
```

Comment:
By using a normal `function` instead of an arrow function, you're creating a new context. Therefore, `this.defaultConfig` will be a property of `configure` instead of `OAuthProvider`, so any access to `this.defaultConfig` outside `configure` will be undefined. I would suggest to update this line as:
```js
this.configure = (params) => {
  this.defaultConfig = sanitizeConfigParams(params);
}
```